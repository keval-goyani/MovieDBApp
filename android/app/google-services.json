{
  "project_info": {
    "project_number": "89996564819",
    "project_id": "moviedb-2908d",
    "storage_bucket": "moviedb-2908d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:89996564819:android:e942630a87714b1d3ea9a6",
        "android_client_info": {
          "package_name": "com.moviedbapp"
        }
      },
      "oauth_client": [
        {
          "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBYrM1MC9Go35l8ra748zqBtCRDkrB_Nn4"
        }
      ],
      "services": {
        "appinvite_service": {
          "other_platform_oauth_client": [
            {
              "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
              "client_type": 3
            }
          ]
        }
      }
    }
  ],
  "configuration_version": "1"
}