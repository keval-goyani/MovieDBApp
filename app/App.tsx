import React, {useEffect} from 'react';
import {SafeAreaView, Text} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import {styles} from './theme';

const App = () => {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <Text>Hello World</Text>
    </SafeAreaView>
  );
};

export default App;
