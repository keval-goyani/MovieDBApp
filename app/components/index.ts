import ChatHeader from './ChatHeader';
import ChatInput from './ChatInput';
import CustomSkeleton from './CustomSkeleton';
import DropDownMenu from './DropDownMenu';
import Header from './Header';
import List from './List';
import ListContainer, { movieListItem } from './ListContainer';
import Loader from './Loader';
import LoadingState from './LoadingState';
import Message from './Message';
import MessageList from './MessageList';
import Movies from './Movies';
import MovieTrailer, { trailerListItem } from './MovieTrailers';
import SearchModal from './SearchModal';
import SkeletonCard from './SkeletonCard';
import TabBarIcon from './TabBarIcon';
import Trailers from './Trailers';
import UserListEmpty from './UserListEmpty';
import UsersList from './UsersList';

export {
  ChatHeader,
  ChatInput,
  DropDownMenu,
  Header,
  List,
  ListContainer,
  MovieTrailer,
  Message,
  MessageList,
  TabBarIcon,
  SearchModal,
  Loader,
  LoadingState,
  Movies,
  movieListItem,
  Trailers,
  trailerListItem,
  UsersList,
  SkeletonCard,
  CustomSkeleton,
  UserListEmpty,
};
