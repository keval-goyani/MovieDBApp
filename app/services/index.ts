import { getPopularMovieData } from './Api';
import immutablePersistenceTransform from './immutablePersistenceTransform';
import {
  alertMessage,
  apiConfig,
  loginError,
  chatCreation,
  clearChat,
  conversationIdCreation,
  convertToTimestamp,
  decryptData,
  encryptData,
  getChatTime,
  getDetails,
  getError,
  getUniqueMovies,
  handleCameraPermission,
  handleDocumentPermission,
  handleGalleryPermission,
  openDocument,
  searchMovie,
  signUpError,
  sortString,
  timestampToTime,
  alertBox,
  storeImageToStorage,
  groupCreation,
  dateConvertor,
  clearConversation,
  sub,
  handleLogOut,
  chatImageHandler,
} from './Utils';

export {
  immutablePersistenceTransform,
  alertMessage,
  apiConfig,
  getDetails,
  getError,
  getPopularMovieData,
  searchMovie,
  getUniqueMovies,
  getChatTime,
  timestampToTime,
  sortString,
  handleGalleryPermission,
  handleCameraPermission,
  clearChat,
  chatCreation,
  handleDocumentPermission,
  conversationIdCreation,
  openDocument,
  encryptData,
  decryptData,
  convertToTimestamp,
  loginError,
  signUpError,
  alertBox,
  storeImageToStorage,
  groupCreation,
  dateConvertor,
  clearConversation,
  sub,
  handleLogOut,
  chatImageHandler,
};
