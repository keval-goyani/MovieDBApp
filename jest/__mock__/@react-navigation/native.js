export const navigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
  replace: jest.fn(),
  popToTop: jest.fn(),
  addListener: jest.fn().mockImplementation((str, cb) => cb()),
  pop: jest.fn(),
  getParam: jest.fn(),
  push: jest.fn(),
};

const useRoute = jest.fn(val => val);
// const useFocusEffect = jest.fn();
const useIsFocused = jest.fn(val => val);
const useNavigation = () => navigation;
const setFocus = jest.fn();
const createNavigationContainerRef = jest.fn();
const DefaultTheme = {};

const CommonActions = { reset: jest.fn() };
const StackActions = {
  pop: jest.fn(),
};

export {
  useRoute,
  useNavigation,
  // useFocusEffect,
  useIsFocused,
  setFocus,
  createNavigationContainerRef,
  DefaultTheme,
  CommonActions,
  StackActions,
};
