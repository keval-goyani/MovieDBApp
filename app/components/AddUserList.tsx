import { useFocusEffect, useNavigation } from '@react-navigation/native';
import React, { useEffect, useState } from 'react';
import { FlatList, Text, TouchableOpacity, View } from 'react-native';
import Check from 'react-native-vector-icons/AntDesign';
import { useDispatch, useSelector } from 'react-redux';
import { asMutable } from 'seamless-immutable';
import { ProfileImage, UserListEmpty } from '../components';
import {
  AddUserListProps,
  NavigationDataType,
  navigationStrings,
  RenderItemTypes,
  SelectedUsersProps,
  strings,
  UserDataType,
} from '../constants';
import { authDataSelectors } from '../redux/AuthRedux';
import { userListSelector } from '../redux/UserListRedux';
import { conversationIdCreation, sub } from '../services';
import { Color, moderateScale } from '../theme';
import { styles } from './styles/AddUserListStyles';

const AddUsersList = ({ userListData, setSelectedUsers }: AddUserListProps) => {
  const navigation: NavigationDataType = useNavigation();
  const dispatch = useDispatch();
  const { user } = useSelector(authDataSelectors.getData);
  const { fetchingUserList } = useSelector(userListSelector.getData);
  const [selectedItems, setSelectedItems] = useState<UserDataType[]>([]);

  useFocusEffect(() => {
    sub({ user, dispatch });
    return () => sub;
  });

  useEffect(() => {
    setSelectedUsers(selectedItems);
  }, [selectedItems, setSelectedUsers]);

  const renderUserList = ({ item, index }: RenderItemTypes) => {
    const usersEmail = [user?.email ?? '', item?.email];
    const conversationId: string = conversationIdCreation(usersEmail);

    const handleOnPress = (selectedUser: UserDataType) => {
      console.log(selectedItems, '<==selectedItems.length');
      console.log(selectedUser, '<==selectedUser');

      if (selectedItems.length) {
        return selectUsers(selectedUser);
      }
      navigation.goBack();
      navigation.navigate(navigationStrings.Chat, {
        conversationId,
        username: item?.username,
        receiverId: item?.uid,
        userStatus: item?.status,
        profileImage: item?.profileImage,
        userEmail: item?.email,
      });
    };

    const getSelected = (usersData: SelectedUsersProps) => {
      return JSON.stringify(selectedItems).includes(JSON.stringify(usersData));
    };

    const selected = getSelected(item);

    const selectUsers = (selectedUser: UserDataType) => {
      if (
        JSON.stringify(selectedItems).includes(JSON.stringify(selectedUser))
      ) {
        const filteredList = selectedItems?.filter(
          (items: UserDataType) => items !== selectedUser,
        );
        setSelectedItems(filteredList);
      } else {
        setSelectedItems([...selectedItems, selectedUser]);
      }
    };

    return (
      <TouchableOpacity
        style={styles.listItem}
        onPress={() => handleOnPress(item)}x
        activeOpacity={0.8}
        onLongPress={() => selectUsers(item)}
        testID={`button-for-test${index}`}>
        <View style={styles.itemContainer}>
          <ProfileImage
            profileImage={item?.profileImage}
            userStatus={item?.status}
          />
          <View style={styles.nameView}>
            <Text style={styles.text}>{item?.username}</Text>
            <Text
              style={styles.lastChatText}
              ellipsizeMode="tail"
              numberOfLines={1}>
              {item?.email}
            </Text>
          </View>
        </View>
        {selected && (
          <>
            <View style={styles.overlay} />
            <Check
              testID="check"
              name={strings.checkIcon}
              size={moderateScale(23)}
              color={Color.lightBlue}
              style={styles.iconStyle}
            />
          </>
        )}
      </TouchableOpacity>
    );
  };

  return (
    <FlatList
      testID={'flatList'}
      data={asMutable(userListData)}
      renderItem={renderUserList}
      ListEmptyComponent={
        <UserListEmpty
          fetching={fetchingUserList}
          userListLength={userListData.length}
        />
      }
      showsVerticalScrollIndicator={false}
      bounces={false}
    />
  );
};

export default AddUsersList;
