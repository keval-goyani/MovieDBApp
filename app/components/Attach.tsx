import { useNavigation } from '@react-navigation/native';
import React, { FC } from 'react';
import { Alert, View } from 'react-native';
import SystemSetting from 'react-native-system-setting';
import { CustomIconRounder } from '../components';
import {
  AttachDataType,
  NavigationDataType,
  navigationStrings,
  strings,
} from '../constants';
import {
  handleCameraPermission,
  handleDocumentPermission,
  handleGalleryPermission,
} from '../services';
import { Icons } from '../theme';
import styles from './styles/AttachStyles';

const Attach: FC<AttachDataType> = ({
  setIsAttach,
  setImagePath,
  setDocumentData,
  conversationId,
  username,
  receiverId,
}) => {
  const navigation: NavigationDataType = useNavigation();

  const turnOnLocation = (enable: boolean) => {
    if (enable) {
      navigation.navigate(navigationStrings.Location, {
        isFromChat: false,
        ...{ conversationId, username, receiverId },
      });
      setIsAttach(false);
    } else {
      Alert.alert(strings.turnOn, strings.goToSetting, [
        {
          onPress: () => SystemSetting.switchLocation(),
          text: strings.setting,
        },
        { text: strings.cancel },
      ]);
    }
  };

  const locationHandler = () => {
    SystemSetting.isLocationEnabled()
      .then(enable => {
        turnOnLocation(enable);
      })
      .catch(error => Alert.alert(strings.warning, error));
  };

  return (
    <View style={styles.container}>
      <View style={styles.iconContainer}>
        <CustomIconRounder
          backgroundColor={styles.documentBackgroundColor}
          path={Icons.fileIcon}
          iconName={strings.document}
          tintColor={styles.documentTintColor}
          onPress={() => {
            setIsAttach(false);
            handleDocumentPermission(setDocumentData);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.cameraBackgroundColor}
          path={Icons.cameraIcon}
          iconName={strings.camera}
          tintColor={styles.cameraTintColor}
          onPress={() => {
            handleCameraPermission(setImagePath);
            setIsAttach(false);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.galleryBackgroundColor}
          path={Icons.galleryIcon}
          iconName={strings.gallery}
          tintColor={styles.galleryTintColor}
          onPress={() => {
            handleGalleryPermission(setImagePath);
            setIsAttach(false);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.locationBackgroundColor}
          path={Icons.locationIcon}
          iconName={strings.map}
          tintColor={styles.locationTintColor}
          onPress={locationHandler}
        />
      </View>
    </View>
  );
};

export default Attach;
