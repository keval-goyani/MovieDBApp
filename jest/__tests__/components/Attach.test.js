import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { Alert } from 'react-native';
import SystemSetting from 'react-native-system-setting';
import { Attach } from '../../../app/components';
import * as Services from '../../../app/services/Utils';
import { navigation } from '@react-navigation/native';
import { act } from 'react-test-renderer';

const setIsAttach = jest.fn();
const setDocumentData = jest.fn();
const setImagePath = jest.fn();

describe('Attach', () => {
  it('Attach snapShot', () => {
    const { toJSON } = render(<Attach />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('document icon should press', () => {
    const { getByText } = render(
      <Attach setIsAttach={setIsAttach} setDocumentData={setDocumentData} />,
    );
    const documentButton = getByText('Document');

    const spy = jest
      .spyOn(Services, 'handleDocumentPermission')
      .mockImplementation();
    fireEvent.press(documentButton);
    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(setDocumentData);
    expect(setIsAttach).toHaveBeenCalledWith(false);
  });

  it('camera icon should press', () => {
    const { getByText } = render(
      <Attach setIsAttach={setIsAttach} setImagePath={setImagePath} />,
    );
    const cameraButton = getByText('Camera');
    const spy = jest
      .spyOn(Services, 'handleCameraPermission')
      .mockImplementation();
    fireEvent.press(cameraButton);
    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(setImagePath);
    expect(setIsAttach).toHaveBeenCalledWith(false);
  });

  it('gallery icon should press', () => {
    const { getByText } = render(
      <Attach setIsAttach={setIsAttach} setImagePath={setImagePath} />,
    );
    const galleryButton = getByText('Gallery');
    const spy = jest
      .spyOn(Services, 'handleGalleryPermission')
      .mockImplementation();
    fireEvent.press(galleryButton);
    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(setImagePath);
    expect(setIsAttach).toHaveBeenCalledWith(false);
  });

  it('navigates to location screen on map icon press when location is enabled', async () => {
    const { getByText } = render(
      <Attach
        setIsAttach={setIsAttach}
        conversationId={'5ddf976ba364b169e8981460ff50ea8d'}
        username={'chandani'}
        receiverId={'1234'}
      />,
    );
    const locationButton = getByText('Map');
    SystemSetting.isLocationEnabled.mockImplementation(() =>
      Promise.resolve(true),
    );
    fireEvent.press(locationButton);
    waitFor(() => {
      expect(navigation.navigate).toHaveBeenCalledWith('Location', {
        isFromChat: false,
        conversationId: '5ddf976ba364b169e8981460ff50ea8d',
        receiverId: '1234',
        username: 'chandani',
      });
      expect(setIsAttach).toHaveBeenCalledWith(false);
    });
  });

  it('give alert for turn on location when location is disabled ', () => {
    const { getByText } = render(<Attach />);
    const locationButton = getByText('Map');
    SystemSetting.isLocationEnabled.mockImplementation(() =>
      Promise.resolve(false),
    );
    const spy = jest.spyOn(Alert, 'alert').mockImplementation();
    fireEvent.press(locationButton);
    Alert.alert = jest
      .fn()
      .mockImplementationOnce((t, m, a, o) => a && a[0] && a[0]?.onPress());
    waitFor(() => {
      expect(spy).toHaveBeenCalled();
    });
  });

  it('should give the alert when SystemSetting.isLocationEnabled is not enable ', () => {
    const { getByText } = render(<Attach />);
    const locationButton = getByText('Map');
    const spy = jest.spyOn(Alert, 'alert').mockImplementation();
    SystemSetting.isLocationEnabled.mockImplementation(() =>
      Promise.reject('error'),
    );
    fireEvent.press(locationButton);
    waitFor(() => {
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });
});
