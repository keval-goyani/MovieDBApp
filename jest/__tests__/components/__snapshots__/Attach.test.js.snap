// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attach Attach snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "bottom": undefined,
      "flex": 1,
      "flexDirection": "row",
      "position": "absolute",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingTop": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginVertical": undefined,
        }
      }
      testID="custom-icon-rounder"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#F9EDFF",
            "borderRadius": 23,
            "opacity": 1,
            "padding": undefined,
          }
        }
        testID="icon-button"
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/assets/icons/fileIcon.png",
            }
          }
          style={
            Array [
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "tintColor": "#642CB8",
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#0d253f",
            "fontSize": undefined,
            "marginTop": undefined,
          }
        }
      >
        Document
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginVertical": undefined,
        }
      }
      testID="custom-icon-rounder"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#04c6ff1a",
            "borderRadius": 23,
            "opacity": 1,
            "padding": undefined,
          }
        }
        testID="icon-button"
      >
        <Image
          source={
            Object {
              "default": Object {
                "attach": Object {
                  "testUri": "../../../app/assets/icons/attach.png",
                },
                "avatar": Object {
                  "testUri": "../../../app/assets/icons/user.png",
                },
                "backIcon": Object {
                  "testUri": "../../../app/assets/icons/backIcon.png",
                },
                "camera": Object {
                  "testUri": "../../../app/assets/icons/camera.png",
                },
                "cameraIcon": Object {
                  "testUri": "../../../app/assets/icons/cameraIcon.png",
                },
                "chatBackground": Object {
                  "testUri": "../../../app/assets/icons/chatBackground.png",
                },
                "close": Object {
                  "testUri": "../../../app/assets/icons/close.png",
                },
                "comingSoon": Object {
                  "testUri": "../../../app/assets/icons/comingSoon.png",
                },
                "communityIcon": Object {
                  "testUri": "../../../app/assets/icons/communityIcon.png",
                },
                "document": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "documentIcon": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "dotMenu": Object {
                  "testUri": "../../../app/assets/icons/dots.png",
                },
                "expandIcon": Object {
                  "testUri": "../../../app/assets/icons/expandIcon.png",
                },
                "fileIcon": Object {
                  "testUri": "../../../app/assets/icons/fileIcon.png",
                },
                "fullStop": Object {
                  "testUri": "../../../app/assets/icons/fullStop.png",
                },
                "galleryIcon": Object {
                  "testUri": "../../../app/assets/icons/galleryIcon.png",
                },
                "homeIcon": Object {
                  "testUri": "../../../app/assets/icons/homeIcon.png",
                },
                "imageDocument": Object {
                  "testUri": "../../../app/assets/icons/imageIcon.png",
                },
                "location": Object {
                  "testUri": "../../../app/assets/icons/share.png",
                },
                "locationIcon": Object {
                  "testUri": "../../../app/assets/icons/locationIcon.png",
                },
                "logout": Object {
                  "testUri": "../../../app/assets/icons/logout.png",
                },
                "menuIcon": Object {
                  "testUri": "../../../app/assets/icons/menuIcon.png",
                },
                "movieDbIcon": Object {
                  "testUri": "../../../app/assets/icons/movieDBIcon.png",
                },
                "movieDbIcon2x": Object {
                  "testUri": "../../../app/assets/icons/movieDB2x.png",
                },
                "movieDbIcon3x": Object {
                  "testUri": "../../../app/assets/icons/movieDB3x.png",
                },
                "moviesIcon": Object {
                  "testUri": "../../../app/assets/icons/moviesIcon.png",
                },
                "noItemFound": Object {
                  "testUri": "../../../app/assets/icons/noItemFound.png",
                },
                "notFound": Object {
                  "testUri": "../../../app/assets/icons/notFoundIcon.png",
                },
                "pdfDocument": Object {
                  "testUri": "../../../app/assets/icons/pdfDocument 2.png",
                },
                "playIcon": Object {
                  "testUri": "../../../app/assets/icons/playIcon.png",
                },
                "plus": Object {
                  "testUri": "../../../app/assets/icons/plusSign.png",
                },
                "pptDocument": Object {
                  "testUri": "../../../app/assets/icons/pptDocument.png",
                },
                "searchIcon": Object {
                  "testUri": "../../../app/assets/icons/searchIcon.png",
                },
                "send": Object {
                  "testUri": "../../../app/assets/icons/send.png",
                },
                "threeDotIcon": Object {
                  "testUri": "../../../app/assets/icons/threeDotIcon.png",
                },
                "trailerIcon": Object {
                  "testUri": "../../../app/assets/icons/trailerIcon.png",
                },
                "trending": Object {
                  "testUri": "../../../app/assets/icons/trending.png",
                },
                "tv": Object {
                  "testUri": "../../../app/assets/icons/tv.png",
                },
                "videoDocument": Object {
                  "testUri": "../../../app/assets/icons/mp4Document.png",
                },
              },
            }
          }
          style={
            Array [
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "tintColor": "#09B2E6",
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#0d253f",
            "fontSize": undefined,
            "marginTop": undefined,
          }
        }
      >
        Camera
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginVertical": undefined,
        }
      }
      testID="custom-icon-rounder"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#EBF4FF",
            "borderRadius": 23,
            "opacity": 1,
            "padding": undefined,
          }
        }
        testID="icon-button"
      >
        <Image
          source={
            Object {
              "default": Object {
                "attach": Object {
                  "testUri": "../../../app/assets/icons/attach.png",
                },
                "avatar": Object {
                  "testUri": "../../../app/assets/icons/user.png",
                },
                "backIcon": Object {
                  "testUri": "../../../app/assets/icons/backIcon.png",
                },
                "camera": Object {
                  "testUri": "../../../app/assets/icons/camera.png",
                },
                "cameraIcon": Object {
                  "testUri": "../../../app/assets/icons/cameraIcon.png",
                },
                "chatBackground": Object {
                  "testUri": "../../../app/assets/icons/chatBackground.png",
                },
                "close": Object {
                  "testUri": "../../../app/assets/icons/close.png",
                },
                "comingSoon": Object {
                  "testUri": "../../../app/assets/icons/comingSoon.png",
                },
                "communityIcon": Object {
                  "testUri": "../../../app/assets/icons/communityIcon.png",
                },
                "document": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "documentIcon": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "dotMenu": Object {
                  "testUri": "../../../app/assets/icons/dots.png",
                },
                "expandIcon": Object {
                  "testUri": "../../../app/assets/icons/expandIcon.png",
                },
                "fileIcon": Object {
                  "testUri": "../../../app/assets/icons/fileIcon.png",
                },
                "fullStop": Object {
                  "testUri": "../../../app/assets/icons/fullStop.png",
                },
                "galleryIcon": Object {
                  "testUri": "../../../app/assets/icons/galleryIcon.png",
                },
                "homeIcon": Object {
                  "testUri": "../../../app/assets/icons/homeIcon.png",
                },
                "imageDocument": Object {
                  "testUri": "../../../app/assets/icons/imageIcon.png",
                },
                "location": Object {
                  "testUri": "../../../app/assets/icons/share.png",
                },
                "locationIcon": Object {
                  "testUri": "../../../app/assets/icons/locationIcon.png",
                },
                "logout": Object {
                  "testUri": "../../../app/assets/icons/logout.png",
                },
                "menuIcon": Object {
                  "testUri": "../../../app/assets/icons/menuIcon.png",
                },
                "movieDbIcon": Object {
                  "testUri": "../../../app/assets/icons/movieDBIcon.png",
                },
                "movieDbIcon2x": Object {
                  "testUri": "../../../app/assets/icons/movieDB2x.png",
                },
                "movieDbIcon3x": Object {
                  "testUri": "../../../app/assets/icons/movieDB3x.png",
                },
                "moviesIcon": Object {
                  "testUri": "../../../app/assets/icons/moviesIcon.png",
                },
                "noItemFound": Object {
                  "testUri": "../../../app/assets/icons/noItemFound.png",
                },
                "notFound": Object {
                  "testUri": "../../../app/assets/icons/notFoundIcon.png",
                },
                "pdfDocument": Object {
                  "testUri": "../../../app/assets/icons/pdfDocument 2.png",
                },
                "playIcon": Object {
                  "testUri": "../../../app/assets/icons/playIcon.png",
                },
                "plus": Object {
                  "testUri": "../../../app/assets/icons/plusSign.png",
                },
                "pptDocument": Object {
                  "testUri": "../../../app/assets/icons/pptDocument.png",
                },
                "searchIcon": Object {
                  "testUri": "../../../app/assets/icons/searchIcon.png",
                },
                "send": Object {
                  "testUri": "../../../app/assets/icons/send.png",
                },
                "threeDotIcon": Object {
                  "testUri": "../../../app/assets/icons/threeDotIcon.png",
                },
                "trailerIcon": Object {
                  "testUri": "../../../app/assets/icons/trailerIcon.png",
                },
                "trending": Object {
                  "testUri": "../../../app/assets/icons/trending.png",
                },
                "tv": Object {
                  "testUri": "../../../app/assets/icons/tv.png",
                },
                "videoDocument": Object {
                  "testUri": "../../../app/assets/icons/mp4Document.png",
                },
              },
            }
          }
          style={
            Array [
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "tintColor": "#2B23C6",
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#0d253f",
            "fontSize": undefined,
            "marginTop": undefined,
          }
        }
      >
        Gallery
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginVertical": undefined,
        }
      }
      testID="custom-icon-rounder"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FFEBF8",
            "borderRadius": 23,
            "opacity": 1,
            "padding": undefined,
          }
        }
        testID="icon-button"
      >
        <Image
          source={
            Object {
              "default": Object {
                "attach": Object {
                  "testUri": "../../../app/assets/icons/attach.png",
                },
                "avatar": Object {
                  "testUri": "../../../app/assets/icons/user.png",
                },
                "backIcon": Object {
                  "testUri": "../../../app/assets/icons/backIcon.png",
                },
                "camera": Object {
                  "testUri": "../../../app/assets/icons/camera.png",
                },
                "cameraIcon": Object {
                  "testUri": "../../../app/assets/icons/cameraIcon.png",
                },
                "chatBackground": Object {
                  "testUri": "../../../app/assets/icons/chatBackground.png",
                },
                "close": Object {
                  "testUri": "../../../app/assets/icons/close.png",
                },
                "comingSoon": Object {
                  "testUri": "../../../app/assets/icons/comingSoon.png",
                },
                "communityIcon": Object {
                  "testUri": "../../../app/assets/icons/communityIcon.png",
                },
                "document": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "documentIcon": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "dotMenu": Object {
                  "testUri": "../../../app/assets/icons/dots.png",
                },
                "expandIcon": Object {
                  "testUri": "../../../app/assets/icons/expandIcon.png",
                },
                "fileIcon": Object {
                  "testUri": "../../../app/assets/icons/fileIcon.png",
                },
                "fullStop": Object {
                  "testUri": "../../../app/assets/icons/fullStop.png",
                },
                "galleryIcon": Object {
                  "testUri": "../../../app/assets/icons/galleryIcon.png",
                },
                "homeIcon": Object {
                  "testUri": "../../../app/assets/icons/homeIcon.png",
                },
                "imageDocument": Object {
                  "testUri": "../../../app/assets/icons/imageIcon.png",
                },
                "location": Object {
                  "testUri": "../../../app/assets/icons/share.png",
                },
                "locationIcon": Object {
                  "testUri": "../../../app/assets/icons/locationIcon.png",
                },
                "logout": Object {
                  "testUri": "../../../app/assets/icons/logout.png",
                },
                "menuIcon": Object {
                  "testUri": "../../../app/assets/icons/menuIcon.png",
                },
                "movieDbIcon": Object {
                  "testUri": "../../../app/assets/icons/movieDBIcon.png",
                },
                "movieDbIcon2x": Object {
                  "testUri": "../../../app/assets/icons/movieDB2x.png",
                },
                "movieDbIcon3x": Object {
                  "testUri": "../../../app/assets/icons/movieDB3x.png",
                },
                "moviesIcon": Object {
                  "testUri": "../../../app/assets/icons/moviesIcon.png",
                },
                "noItemFound": Object {
                  "testUri": "../../../app/assets/icons/noItemFound.png",
                },
                "notFound": Object {
                  "testUri": "../../../app/assets/icons/notFoundIcon.png",
                },
                "pdfDocument": Object {
                  "testUri": "../../../app/assets/icons/pdfDocument 2.png",
                },
                "playIcon": Object {
                  "testUri": "../../../app/assets/icons/playIcon.png",
                },
                "plus": Object {
                  "testUri": "../../../app/assets/icons/plusSign.png",
                },
                "pptDocument": Object {
                  "testUri": "../../../app/assets/icons/pptDocument.png",
                },
                "searchIcon": Object {
                  "testUri": "../../../app/assets/icons/searchIcon.png",
                },
                "send": Object {
                  "testUri": "../../../app/assets/icons/send.png",
                },
                "threeDotIcon": Object {
                  "testUri": "../../../app/assets/icons/threeDotIcon.png",
                },
                "trailerIcon": Object {
                  "testUri": "../../../app/assets/icons/trailerIcon.png",
                },
                "trending": Object {
                  "testUri": "../../../app/assets/icons/trending.png",
                },
                "tv": Object {
                  "testUri": "../../../app/assets/icons/tv.png",
                },
                "videoDocument": Object {
                  "testUri": "../../../app/assets/icons/mp4Document.png",
                },
              },
            }
          }
          style={
            Array [
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "tintColor": "#ff0080",
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#0d253f",
            "fontSize": undefined,
            "marginTop": undefined,
          }
        }
      >
        Map
      </Text>
    </View>
  </View>
</View>
`;
